packages/components/nodes/sequentialagents/LLMNode/LLMNode.ts

```diff
@@ -535,7 +543,7 @@ async function agentNode(
         }
 
         // @ts-ignore
-        state.messages = restructureMessages(llm, state)
+        state.messages = nodeData.inputs?.disableConversationHistory | false ? [] : restructureMessages(llm, state)
```

Comment:
I dont we can pass empty array, perhaps we need to take the last message: 
![image](https://github.com/user-attachments/assets/c7c9485a-afa8-45ff-8aad-ac3bf52d7449)

Reply:
> I don't think we can pass an empty array; perhaps we need to take the last message  

Based on my perspective and local tests, it doesn’t seem necessary to always pass the last message. However, we do need to ensure that at least one of the following is included: the system prompt, human prompt, or message history code. Otherwise, the following error is raised:  

```
Error buildAgentGraph - Error: 400 {"error":{"message":"'messages' : minimum number of items is 1","type":"invalid_request_error"}}
```

Passing only the latest history message undermines the purpose of this feature, which is to provide the user full control over the messages passed to the node. Instead, I believe that we should allow users to explicitly define the messages to be sent.

Reply:
Right, Im thinking of whats the best way to cater to all 3 scenarios:

- taking the last message from previous node as input (i.e: `state.messages[-1]`)
- taking the user question as input
- taking all the conversation state messages (default)

So I think the best way to allow this is to have an Option field where user can select which one to be used.

Reply:
Thank you for the insightful suggestion! I've implemented the changes as you recommended and conducted some local tests, which indicate that everything is functioning as expected. Please review the current implementation and let me know if it meets your expectations or if there are any further adjustments you'd like to see.

Reply:
Here's a langsmith trace of one my workflow test: [https://smith.langchain.com/public/c04e60c1-0659-46a5-bf8f-eefb5ed312df/r](https://smith.langchain.com/public/c04e60c1-0659-46a5-bf8f-eefb5ed312df/r)

The workflow includes 4 LLM nodes: `jokeempty`, `jokeuserquestion`, `jokelastmessage`, and `jokefullhistory`. It also features an agent with a calculator, which receives only the user question.